### ISSUE TITLE
REST API Server Example Project Availability?

### ISSUE THREAD

**Issue by geoah** [sogko/slumber#1]
This seems like an amazing project, thank you for sharing it with us.

On the `README.md` there is a checked note...
> Create a REST API server project using this package as a boilerplate without changing this package

Is there such a project available publicly? I would really be interested in how you implemented it.
Thanks

**Comment by sogko** [comment]
Hey @geoah 
Thanks a lot for the interest in the project!

I have a couple of REST-ful projects that imported this package, though they are in private repos currently. I was planning on publishing a public example project, though I can't promise when exactly I can get it out of the pipeline.

But if you are interested right now, you can take a look at the main.go file , which basically 

- load API routes
- load ACL maps
- create configurations for the server middlewares
- create a server instance
- setup routes for server instance
- run server

The project is still pretty raw and is still in its infant stage, teething problems abound.
(Might need a proper package name that isn't 30 characters long lol)
But I would love getting input and feedback on it.

I'll update this issue when I do make plans to properly publish an example project using this package.

Thanks again!

**Comment by sogko** [comment]
Hey @geoah 

I've just published an example project using this project (renamed as `slumber`) as an imported package.
It's a REST API server that has 1 REST resource (`books`), along with the 2 built-in resources (`users` and `sessions`)

The project is available at: https://github.com/sogko/slumber-books-api-example

Currently I have yet to write documentation for this, I plan to work on it over the weekend and hopefully it'll be up in a few days.

Let me know if you need more help and any feedback would be great!

Thanks again

**Comment by geoah** [comment]
Hey @sogko 

Thanks a million for this example project,
I've started using - woa I just noticed that you renamed the project - slumber in a project myself so I'll be providing more feedback quite soon.

+1 for the name change, it will help a lot.
I really like the project structure and the way everything works.

I do understand that this project is mongodb specific but if it would be possible to have the `users`, `sessions`, `tokens` optionally included, slumber could server as a base for a whole lot more projects.

eg. I'm using rethinkdb via `gorething` and had to mess up quite a lot of files to make it work. Also `server/server.go` could be a bit more modular as to the database init.

Additionally my users implementation does not have a username which means that the default `users`, `sessions` and `tokens` go out of the window. I don't mind that but their files are spread beyond the `users` and `sessions` folders. A better grouping of the files might help not having to fork the lib.

I'll try to document some things a bit better and as I start to better understand how everything works make some suggestions and try to see how things could get more decoupled.

Thank you for sharing this.
I'll be using this quite a bit :)
